config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_vertexaiendpoints.vertexai.cnrm.cloud.google.com.yaml

```diff
@@ -162,233 +207,669 @@ spec:
+                                description: The maximum number of replicas this DeployedModel
+                                  may be deployed on when the traffic against it increases.
+                                  If the requested value is too large, the deployment
+                                  will error, but if deployment succeeds then the
+                                  ability to scale the model to that many replicas
+                                  is guaranteed (barring service outages). If traffic
+                                  against the DeployedModel increases beyond what
+                                  its replicas at maximum may handle, a portion of
+                                  the traffic will be dropped. If this value is not
+                                  provided, will use min_replica_count as the default
+                                  value. The value of this field impacts the charge
+                                  against Vertex CPU and GPU quotas. Specifically,
+                                  you will be charged for max_replica_count * number
+                                  of cores in the selected machine type) and (max_replica_count
+                                  * number of GPUs per replica in the selected machine
+                                  type).
+                                type: integer
+                              minReplicaCount:
+                                description: The minimum number of machine replicas
+                                  this DeployedModel will be always deployed on. This
+                                  value must be greater than or equal to 1. If traffic
+                                  against the DeployedModel increases, it may dynamically
+                                  be deployed onto more replicas, and as traffic decreases,
+                                  some of these extra replicas may be freed.
+                                type: integer
+                            type: object
+                          type: array
+                        displayName:
+                          description: The display name of the DeployedModel. If not
+                            provided upon creation, the Model's display_name is used.
+                          type: string
+                        enableAccessLogging:
+                          description: These logs are like standard server access
+                            logs, containing information like timestamp and latency
+                            for each prediction request. Note that Stackdriver logs
+                            may incur a cost, especially if your project receives
+                            prediction requests at a high queries per second rate
+                            (QPS). Estimate your costs before enabling this option.
+                          type: boolean
+                        enableContainerLogging:
```

Comment:
Similarly, I think this field (and a few others) is input-only.  Do we want to wait till we understand it better before exposing it here?  There is some use-case for the actual state of the object if we were unable to apply changes, but I'm not sure.

Reply:
Do you mean the field is output-only? The API generates them but users cannot configure them. I am not sure I fully understood your concern.

Reply:
I dug into the API a bit, and looks like while [`deployedModels` is an "output-only" field](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.endpoints#resource:-endpoint), there is a method to ["mutate" each individual deployed model](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.endpoints/mutateDeployedModel). There are also methods to [add](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.endpoints/deployModel#google.cloud.aiplatform.v1beta1.EndpointService.DeployModel) or [remove](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.endpoints/undeployModel) deployed models in this list.

Looking at the basic use case here: https://cloud.google.com/vertex-ai/docs/general/deployment#aiplatform_deploy_model_custom_trained_model_sample-drest, the endpoints are used to deploy the models, and I think the deployed model should be a separate resource instead of being captured here.

Reply:
> and I think the deployed model should be a separate resource instead of being captured here.

Could you clarify? Are you suggesting that we should remove `deployedModels` from the `status.observedState`?

Reply:
Ah, yes, I think I understand what's going on now.  So I did mean that `enableAccessLogging` and `enableContainerLogging` looked like input-only fields.   It's because the deployed models are more list a subresource, and this resource isn't following our normal API design.

I don't know if we should actually split these into their own resources (as we normally would for subresources), but they are _not_ really output-only fields, so they likely don't belong in status.

I also don't think this resource is useful without being able to deploy Models, so I don't think it makes sense to merge this resource until we've figured this out... which is a bummer in terms of timing.

Reply:
> Are you suggesting that we should remove deployedModels from the status.observedState?

Yes, I think it's a fine-grained resource that should not be represented here.

> I don't know if we should actually split these into their own resources (as we normally would for subresources), but they are not really output-only fields, so they likely don't belong in status.

I consider "resource deployment" a reasonable use case to be represented individually and declaratively. We should reach out to the API team to understand more details. We probably have to "redo the API design" for them in the KCC resource.

> I also don't think this resource is useful without being able to deploy Models, so I don't think it makes sense to merge this resource until we've figured this out... 

Agreed. But it's always doable to release the resource without the uncertain fields (i.e. we can ignore the tricky deployedModels field) and get feedback from users while we are finalizing the design.